LEAF_Request_Portal/form.php

```diff
@@ -3051,135 +3057,142 @@ public function query($inQuery)
-                        AND records_workflow_state.stepID IS NULL
-                        AND submitted > 0
-                        AND deleted = 0', array());
-            foreach ($res2 as $item)
+            if($joinRecordResolutionData)
             {
-                if($data[$item['recordID']]['recordResolutionData']['fulfillmentTime'] == null
-                    || $data[$item['recordID']]['recordResolutionData']['fulfillmentTime'] < $item['fulfillmentTime']) {
-                    $data[$item['recordID']]['recordResolutionData']['lastStatus'] = $item['lastStatus'];
-                    $data[$item['recordID']]['recordResolutionData']['fulfillmentTime'] = $item['fulfillmentTime'];
+                $res2 = $this->db->prepared_query('SELECT recordID, lastStatus, records_step_fulfillment.stepID, fulfillmentTime FROM records
+                        LEFT JOIN records_step_fulfillment USING (recordID)
+                        LEFT JOIN records_workflow_state USING (recordID)
+                        WHERE recordID IN (' . $recordIDs . ')
+                            AND records_workflow_state.stepID IS NULL
+                            AND submitted > 0
+                            AND deleted = 0', array());
+                foreach ($res2 as $item)
+                {
+                    if($data[$item['recordID']]['recordResolutionData']['fulfillmentTime'] == null
+                        || $data[$item['recordID']]['recordResolutionData']['fulfillmentTime'] < $item['fulfillmentTime']) {
+                        $data[$item['recordID']]['recordResolutionData']['lastStatus'] = $item['lastStatus'];
+                        $data[$item['recordID']]['recordResolutionData']['fulfillmentTime'] = $item['fulfillmentTime'];
+                    }
                 }
             }
-        }
 
-        if ($joinRecordResolutionBy === true) {
-            require_once 'VAMC_Directory.php';
-            $dir = new VAMC_Directory;
+            if ($joinRecordResolutionBy === true) {
+                require_once 'VAMC_Directory.php';
+                $dir = new VAMC_Directory;
 
-            $strSQL = "SELECT recordID, action_history.userID as resolvedBy, action_history.stepID, action_history.actionType FROM action_history ".
-                      "LEFT JOIN records USING (recordID) ".
-                      "INNER JOIN workflow_routes USING (stepID) ".
-                      "LEFT JOIN records_workflow_state USING (recordID) ".
-                      "WHERE recordID IN ($recordIDs) ".
-                        "AND action_history.actionType = workflow_routes.actionType ".
-                        "AND records_workflow_state.stepID IS NULL ".
-                        "AND nextStepID = 0 ".
-                        "AND submitted > 0 ".
-                        "AND deleted = 0";
+                $strSQL = "SELECT recordID, action_history.userID as resolvedBy, action_history.stepID, action_history.actionType FROM action_history ".
```

Comment:
I had been of two minds going through the code on this.  Since your going to be making change anyway, please go back and set up all the sql calls you touched to this method:  define the call in a string then use it in the db call like here.